File: codingjs/canBalance.js
function canBalance(nums){
  for (let i = 0; i < nums.length - 1; i++) {
    const firstHalfSum = nums.slice(0, i + 1).reduce((a, c) => a + c);
    const secondHalfSum = nums.slice(i + 1).reduce((a, c) => a + c);
    if (firstHalfSum === secondHalfSum) return true
  }
  return false;
}

File: codingbat/canBalance.java
public boolean canBalance(int[] nums) {
  if (nums.length < 2) return false;
  int leftSum = 0;
  for (int i = 0; i < nums.length - 1; i++) {
    leftSum += nums[i];
    int rightSum = 0;
    for (int j = i + 1; j < nums.length; j++) {
      rightSum += nums[j];
    }
    if (leftSum == rightSum) return true;
  }
  return false;
}

